lib/Doctrine/ORM/Query/Lexer.php

```diff
@@ -126,10 +128,11 @@ public function __construct($input)
     protected function getCatchablePatterns()
     {
         return array(
-            '[a-z_\\\][a-z0-9_\:\\\]*[a-z0-9_]{1}',
-            '(?:[0-9]+(?:[\.][0-9]+)*)(?:e[+-]?[0-9]+)?',
-            "'(?:[^']|'')*'",
-            '\?[0-9]*|:[a-z_][a-z0-9_]*'
+            '[a-z_][a-z0-9_]*\:[a-z_][a-z0-9_]*', // aliased name
+            '\\\?[a-z_][a-z0-9_]*(?:\\\[a-z_][a-z0-9_]*)*', // identifier or qualified name
```

Comment:
It was added because early adopters (thus the low value in DDC ticket) used `\` a lot. We never wanted that though. It gets stripped before even sent anywhere. In DQL things are different, you gotta send it the proper value. =)